src/core/tsi/ssl_transport_security.h

```diff
@@ -178,6 +196,17 @@ void tsi_ssl_client_handshaker_factory_unref(
 typedef struct tsi_ssl_server_handshaker_factory
     tsi_ssl_server_handshaker_factory;
 
+/* Creates a TLS server TSI handshaker.
+  - options: TLS credential options instance.
+  - factory: address of server handshaker factory to be created.
+  - handshaker: address of the handshaker pointer to be created.
+  This method returns TSI_OK on success or TSI_INVALID_PARAMETER in the case
+  where a parameter is invalid. Note that some fields of handshaker will be
+  populated once credential reload is finished. */
+tsi_result tls_tsi_server_handshaker_create(
+    const grpc_tls_credentials_options* options,
+    tsi_ssl_server_handshaker_factory** factory, tsi_handshaker** handshaker);
```

Comment:
This API is confusing because, in some cases, factory object is created during this call, whereas in other cases, the factory object is reused. For example, if caller passes a factory object, this function inside may destroy the factory and replace a new one. 

Same comment on the client side.

I prefer to have a separate create functions. tsi_create_tls_server_handshaker_factory and tsi_tls_server_handshaker_create to be explicit. The credential reload shall be done at spiffe security connector. Keep SSL TSI simple. This would be similar to current credential logic in ssl security connector.

The idea is:
- When you create security connector, you try to create handshaker factory. It may not be possible, if there is no initial key material.
- Every time add handshaker is called, you check credential reload and see if there are new credentials. If so, re-create handshaker factory.
- Then you create handshaker from handshaker factory inside add handshaker.